src/main/java/com/greenfox/lvlup/service/BadgeLevelService.java

```diff
@@ -0,0 +1,40 @@
+package com.greenfox.lvlup.service;
+
+import com.greenfox.lvlup.model.entity.Badge;
+import com.greenfox.lvlup.model.entity.BadgeLevel;
+import com.greenfox.lvlup.model.entity.User;
+import com.greenfox.lvlup.model.entityTestingDto.BadgeDTO;
+import com.greenfox.lvlup.model.entityTestingDto.BadgeLevelDTO;
+import com.greenfox.lvlup.repositrory.BadgeLevelRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@Service
+public class BadgeLevelService {
+    private UserService userService;
+
+    @Autowired
+    public BadgeLevelService(UserService userService) {
+        this.userService = userService;
+    }
+
+    public BadgeLevelDTO getDTOfromBadgeLevel(BadgeLevel badgeLevel) {
+        BadgeLevelDTO badgeLevelDTO = new BadgeLevelDTO();
+        badgeLevelDTO.level = badgeLevel.getLevel();
+        badgeLevelDTO.description = badgeLevel.getDescription();
+        badgeLevelDTO.holders = new HashSet<>();
+        Set<User> usersOfTheBadgeLevel = badgeLevel.getHolders();
```

Comment:
just a naming thing, too: maybe usersHoldingBadgeLevel instead of UsersOfThe... ?

Reply:
Modified name as per Mark's suggestion:
  Set<User> usersHoldingBadgeLevel = badgeLevel.getHolders();